### PR TITLE
Power BI report dashboard owner editor metadata

### PR DESCRIPTION
**Pull request by elic-eon**
### 🤔 Why?

To get more metadata from Power BI APIs

### 🤓 What?

- Add owner/modifier information
- Add endorsement information

### 🧪 Tested?

Unit tests

### CODE REVIEW

File: metaphor/power_bi/extractor.py [hunk 1]
```diff
@@ -506,25 +512,58 @@ def safe_parse_date(datetime_str: Optional[str]) -> Optional[datetime]:
...
 
+    @staticmethod
+    def _safe_parse_ISO8601(iso8061_str: Optional[str]) -> Optional[datetime]:
+        if iso8061_str is None:
+            return None
+        try:
+            return isoparse(iso8061_str).replace(tzinfo=timezone.utc)
+        except Exception:
+            logger.error(f"Failed to parse ISO8061 time: {iso8061_str}")
+            return None
```
Comment (alyiwang) [inline metaphor/power_bi/extractor.py:565]: I think This is generic enough to go to common util

**Review by alyiwang — COMMENTED** [review]
LGTM

File: metaphor/power_bi/extractor.py (hunk 1 above)
Comment (mars-lan) [reply metaphor/power_bi/extractor.py:565]: I vaguely remember writing something similar before but can't seem to find it immediately.

File: metaphor/common/utils.py [hunk 2]
```diff
@@ -47,6 +53,16 @@ def to_utc_time(time: datetime) -> datetime:
     return time.replace(tzinfo=timezone.utc)
 
 
+def safe_parse_ISO8601(iso8061_str: Optional[str]) -> Optional[datetime]:
```
Comment (mars-lan) [inline metaphor/common/utils.py:56]: Add a one-line description?

**Review by mars-lan — DISMISSED** [review]
Otherwise LGTM

**Review by mars-lan — APPROVED** [review]

**Pull request merged**